### ISSUE TITLE
Support flutter build aar

### ISSUE THREAD

**Issue by xster** [flutter/flutter#26099]
I feel like there must be an existing bug but I can't find one. 

One concrete solution/option to https://github.com/flutter/flutter/issues/25979

More or less https://github.com/flutter/flutter/issues/25323 and https://github.com/flutter/flutter/issues/24948

cc @dnfield

**Comment by vvarp** [comment]
Anyone care to share their current best approach for generating easily embeddable aar/framework?

We started work on this about 2 weeks ago and we ended up with such convoluted structure that I am afraid it might break any moment now. On top of this we are seeing quite a lot of issues when using plugins that pull in additional dependencies, and we are going off on patching spree almost every day. 

Right now our setup consist of separate subproject for each platform, with plenty of glue in form of Python scripts extracting bits and pieces from different build dirs and putting together contents for a podspec (iOS), or with dancing around dependencies problem using fat-aar (Android).

I appreciate the fact that we are still in the early days and easy embedding Flutter views in existing apps isn't the most chased/desired feature, but I have a feeling that we're not first ones trying to achieve that, and there might be some unheard success stories!

**Comment by neiljaywarner** [comment]
@vvarp  Please let me know if you come up with anything good. I am also curious when we will not be in the early days... and for enterprise it is a desired feature right?

**Comment by dnfield** [comment]
#29328 would help this

**Comment by xster** [comment]
Here's a sample of what Alibaba did to breakdowns our existing xcode_backend.sh to produce .frameworks

```bash
fwn_repo_path=$(pwd)
export FLUTTER_ROOT="${fwn_repo_path}/flutter"
export FLUTTER_APPLICATION_PATH="${fwn_repo_path}"
export FLUTTER_BUILD_MODE="${build_mode}"
export SOURCE_ROOT="${fwn_repo_path}/ios"
export ARCHS="armv7 arm64"
export SDKROOT="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk"
/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build

# Call LC_LOAD_DYLIB, otherwise it crashes on iOS 9.
if [ "$build_mode" = "release" ];then
    echo "===注入LC_LOAD_DYLIB命令==="
    echo "Injecting LC_LOAD_DYLIB command..."
    sync_tools/insert_dylib /usr/lib/libSystem.B.dylib build/aot/App.framework/App
    rm -rf build/aot/App.framework/App
    cp build/aot/App.framework/App_patched build/aot/App.framework/App
    rm -rf build/aot/App.framework/App_patched
fi
```

cc @jonahwilliams

**Comment by xster** [comment]
Note from @mehmetf, we should take care to cover outputting modules with fat binaries.

**Comment by xster** [comment]
https://github.com/flutter/flutter/issues/22102 is an Android sub-bug

**Comment by xster** [comment]
Although https://github.com/flutter/flutter/issues/22102 is a subset and useful on its own (such as using include_flutter.groovy and being able to access resources) but nevertheless, we'll necessarily solve it while doing this.

**Comment by xster** [comment]
Based on @blasten's feedback, splitting Android and iOS work. iOS is in #36968

I think the remaining work for Android is continuing to monitor the flutter build aar command on master and to update the documentation for add-to-app.

**Comment by truongsinh** [comment]
`flutter build aar` is now available on 1.8.3, should we close this issue?

**Comment by blasten** [comment]
Yeah!
